python/lsst/dax/ppdb/sql/_ppdb_sql.py

```diff
@@ -501,22 +520,29 @@ def store(
             self._store_table_data(sources, connection, update, "DiaSource", 100)
             self._store_table_data(forced_sources, connection, update, "DiaForcedSource", 1000)
 
-    def _store_insert_id(
-        self, replica_chunk: ReplicaChunk, connection: sqlalchemy.engine.Connection, update: bool
+    def upsert(
+        self,
+        connection: Connection,
+        update: bool,
+        table: Table,
+        values: dict[str, Any],
+        row: dict[str, Any],
     ) -> None:
-        """Insert or replace single record in PpdbReplicaChunk table"""
-        # `astropy.Time.datetime` returns naive datetime, even though all
-        # astropy times are in UTC. Add UTC timezone to timestampt so that
-        # database can store a correct value.
-        insert_dt = datetime.datetime.fromtimestamp(
-            replica_chunk.last_update_time.unix_tai, tz=datetime.timezone.utc
-        )
-        now = datetime.datetime.fromtimestamp(astropy.time.Time.now().unix_tai, tz=datetime.timezone.utc)
-
-        table = self._get_table("PpdbReplicaChunk")
+        """Insert or upsert a row depending on `update` flag.
 
-        values = {"last_update_time": insert_dt, "unique_id": replica_chunk.unique_id, "replica_time": now}
-        row = {"apdb_replica_chunk": replica_chunk.id} | values
+        Parameters
+        ----------
+        connection : sqlalchemy.engine.Connection
```

Comment:
```suggestion
        connection : `sqlalchemy.engine.Connection`
```
And for all other types.